src/pages/RFOX/components/AddressSelection.tsx

```diff
@@ -77,6 +83,15 @@ export const AddressSelection: FC<AddressSelectionProps> = ({
     select: selectRuneAddress,
   })
 
+  useEffect(() => {
+    handleRuneAddressChange(undefined)
+    setIsManualAddress(false)
+  }, [stakingAssetAccountId, handleRuneAddressChange])
```

Comment:
If the main account change, reset the states because the users might open the custom address part on a fresh account but the next account may have an active staking already